### PR TITLE
Improve the PR workflow actions

### PR DESCRIPTION
**Pull request by jnidzwetzki**
This patch includes two changes to the PR handling workflow:

(1) It changes the trigger for the workflow to pull_request_target. So,
    PRs can now also be assigned to reviewers if the PR is opened from
    external sources.

(2) A workflow has been added that automatically assigns PRs to the
    author.

<strike>**Note:** Waits for `#eng-security-discuss` feedback before it is converted into a PR ready for review.</strike> (Done)

### CODE REVIEW

File: .github/workflows/pr-handling.yaml
Comment (paulfantom) [inline .github/workflows/pr-handling.yaml:38]: Why not include the content of this script here? Especially since script is just a one-liner starting a docker container.

Comment (jnidzwetzki) [reply .github/workflows/pr-handling.yaml:38]: This line is already part of the existing workflow (see the deleted file `.github/workflows/pull-review.yaml` below). It might be a good refactoring to include it directly. But I think this should be done in another PR (to keep the focus of the PR clear).

Comment (thanasisk) [reply .github/workflows/pr-handling.yaml:38]: 👍  to @paulfantom 's suggestion - given that you will add it as a comment - something that does not modify functionality, I think it is acceptable to include this in this PR.

File: .github/workflows/pr-handling.yaml
Comment (thanasisk) [inline .github/workflows/pr-handling.yaml:14]: From https://securitylab.github.com/research/github-actions-preventing-pwn-requests/
```
pull_request_target runs in the context of the target repository of the PR, rather than in the merge commit. This means the standard checkout action uses the target repository to prevent accidental usage of the user supplied code.
```

Therefore, my understanding is that even if a malicious user modifies `scripts/pull-review.sh` - the original `pull-review.sh` from the target repo will be executed. So that makes it safe for inclusion, to my understanding.

File: .github/workflows/pr-handling.yaml
Comment (thanasisk) [inline .github/workflows/pr-handling.yaml:22]: kindly keep in mind that v1.6.1 is out - however from the Changelog, no security fixes are evident. Please consider keeping this up-2-date with the releases.

**Comment by thanasisk** [comment]
@jnidzwetzki @paulfantom even though from the documentation LGTM, once this is merged please reach out immediately to me, so I can carry out some hands on attacks. Thanks!

File: .github/workflows/pr-handling.yaml
Comment (jnidzwetzki) [reply .github/workflows/pr-handling.yaml]: Thank you for pointing this out. The action is updated.

File: .github/workflows/pr-handling.yaml
Comment (jnidzwetzki) [reply .github/workflows/pr-handling.yaml]: As suggested, I have integrated the script `scripts/pull-review.sh` directly into the workflow.

**Review by akuzm — APPROVED** [review]

**Review by konskov — APPROVED** [review]

**Pull request merged**
